progenitor-client/src/progenitor_client.rs

```diff
@@ -240,9 +240,14 @@ pub enum Error<E = ()> {
     /// A documented, expected error response.
     ErrorResponse(ResponseValue<E>),
 
+    /// An expected response when upgrading connection.
+    InvalidUpgrade(reqwest::Error),
```

Comment:
Let's move this up to be right after `CommunicationError` which is probably what this should have been initially.

```suggestion
    /// An error occurred when upgrading connection.
    UpgradeError(reqwest::Error),
```